lib/compress/zstd_compress.c

```diff
@@ -354,6 +374,23 @@ size_t ZSTD_CCtx_setParameter(ZSTD_CCtx* cctx, ZSTD_cParameter param, unsigned v
         DEBUGLOG(5, " setting overlap with nbThreads == %u", cctx->requestedParams.nbThreads);
         return ZSTD_CCtxParam_setParameter(&cctx->requestedParams, param, value);
 
+    case ZSTD_p_longDistanceMatching:
+        if (cctx->cdict) return ERROR(stage_wrong);
```

Comment:
I understand that long range match is not meant to work in combination with dictionary,

My question is, how bad is it if it happens ?
Does it crashes ? does it produces corrupted data ?
Or is it just inefficient ?

Also : 
what about "one time dictionaries" declared with `ZSTD_refPrefix()` ?

Reply:
It won't crash/produce corrupted data (from some quick testing) with `refPrefix` or cdict. Long distance matching doesn't have any effect except in the compression stage. However, it does modify the window size (which I believe isn't used in loading a dictionary?), and to keep with the semantics of "not modifying compression parameters after loading a dictionary", I didn't allow it to be set if a cdict was loaded.

FYI: There's currently a bug in parameter setting with cdict and `CCtx_loadDictionary` due to the dual use of `copyCCtx` as a function to copy a cctx and to copy a dictionary `refContext`. I believe the `refContext` will only ever contain compression parameters in its `appliedParameters`, so after `ZSTD_CCtx_loadDictionary`, `ZSTD_copyCCtx_internal` is called in `ZSTD_compressBegin_internal` and all other parameters are discarded. I'll send in a fix.